File: code/generate_strategies/RA(method)/python/function_bitarray_extend.py
from bitarray import bitarray

class BitarrayExtender:
    def extend_bits(self, bits: str, values: str) -> str:
        if not isinstance(bits, str): 
            raise TypeError("Input must be a string of '0' and '1'.")
        if not all(v in ('0', '1') for v in values):
            raise ValueError("All elements in the iterable must be '0' or '1'.")
        
        ba = bitarray(bits)
        ba.extend(values)  
        return ba.to01()

File: code/generate_strategies/IR(CoT)/java/java/function_bitarray_extend.java
package com.example;

import org.apache.commons.lang3.StringUtils;

public class BitarrayExtender {
    
    
    
    
    
    public String extend_bits(String bits, String values) {
        
        if (bits == null) {
            throw new IllegalArgumentException("Input must be a string of '0' and '1'.");
        }
        
        if (values == null) {
            throw new IllegalArgumentException("Iterable input must be a string of '0' and '1'.");
        }
        
        for (int i = 0; i < values.length(); i++) {
            char v = values.charAt(i);
            if (v != '0' && v != '1') {
                throw new IllegalArgumentException("All elements in the iterable must be '0' or '1'.");
            }
        }
        
        String result = StringUtils.join(new String[]{bits, values}, "");
        return result;
    }
}